DistributedTransaction/DistributedTransaction.tla

```diff
@@ -381,36 +390,132 @@ ServerCommit ==
 ServerCleanupStaleLock ==
   \E k \in KEY :
     \E l \in key_lock[k] :
-      /\ SendReqs({[type |-> "cleanup",
+      \* the `resolving_pessimistic_lock` field is set to *TRUE*
+      CASE l.type \in {"lock_key"} ->
+        /\ SendReqs({[type |-> "check_txn_status_req",
                     start_ts |-> l.ts,
-                    primary |-> l.primary]})
-      /\ UNCHANGED <<resp_msgs, client_vars, key_vars, next_ts>>
+                    primary |-> l.primary, 
+                    rollback_if_not_exist |-> TRUE,
+                    resolving_pessimistic_lock |-> TRUE
+                    ]})
+        /\ UNCHANGED <<resp_msgs, client_vars, key_vars, next_ts>>
+        \/ /\ SendReqs({[type |-> "check_txn_status_req",
+                    start_ts |-> l.ts,
+                    primary |-> l.primary, 
+                    rollback_if_not_exist |-> FALSE,
+                    resolving_pessimistic_lock |-> TRUE
+                    ]})
+           /\ UNCHANGED <<resp_msgs, client_vars, key_vars, next_ts>>
+        
+      \* the `resolving_pessimistic_lock` field is set to *FALSE*
+      [] l.type \in {"prewrite_optimistic", "prewrite_pessimistic"} ->
+        /\ SendReqs({[type |-> "check_txn_status_req",
+                     start_ts |-> l.ts,
+                     primary |-> l.primary, 
+                     rollback_if_not_exist |-> TRUE,
+                     resolving_pessimistic_lock |-> FALSE
+                     ]})
+        /\ UNCHANGED <<resp_msgs, client_vars, key_vars, next_ts>>
+        \/ /\ SendReqs({[type |-> "check_txn_status_req",
+                    start_ts |-> l.ts,
+                    primary |-> l.primary, 
+                    rollback_if_not_exist |-> FALSE,
+                    resolving_pessimistic_lock |-> FALSE
+                    ]})

```

Comment:
Use `LET IN` on `rollback_if_not_exist` and `resolving_pessimistic_lock` to reduce duplication.